huaweicloud/services/oms/resource_huaweicloud_oms_migration_task_group.go

```diff
@@ -0,0 +1,835 @@
+				Optional: true,
+				ForceNew: true,
+				Default:  true,
+			},
+			"bandwidth_policy": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 5,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"max_bandwidth": {
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+						"start": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"end": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+			},
+			"source_cdn": {
+				Type:     schema.TypeList,
+				Optional: true,
+				ForceNew: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"domain": {
+							Type:     schema.TypeString,
+							Required: true,
+							ForceNew: true,
+						},
+						"protocol": {
+							Type:     schema.TypeString,
+							Required: true,
+							ForceNew: true,
+						},
+						"authentication_type": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+							Default:  "NONE",
+						},
+						"authentication_key": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+						},
+					},
+				},
+			},
+			"status": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"total_time": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"total_num": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"success_num": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"fail_num": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"total_size": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"complete_size": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func buildTaskGroupSrcNodeOpts(rawSrcNode []interface{}) *oms.TaskGroupSrcNode {
+	if len(rawSrcNode) != 1 {
+		return nil
+	}
+	srcNode := rawSrcNode[0].(map[string]interface{})
+
+	srcNodeOpts := oms.TaskGroupSrcNode{
+		CloudType: utils.StringIgnoreEmpty(srcNode["data_source"].(string)),
+		Region:    utils.StringIgnoreEmpty(srcNode["region"].(string)),
+		Ak:        utils.StringIgnoreEmpty(srcNode["access_key"].(string)),
+		Sk:        utils.StringIgnoreEmpty(srcNode["secret_key"].(string)),
+		AppId:     utils.StringIgnoreEmpty(srcNode["app_id"].(string)),
+		Bucket:    utils.StringIgnoreEmpty(srcNode["bucket"].(string)),
+	}
+
+	if srcNode["list_file_bucket"].(string) != "" {
+		srcNodeOpts.ListFile = &oms.ListFile{
+			ObsBucket:   srcNode["list_file_bucket"].(string),
+			ListFileKey: srcNode["list_file_key"].(string),
+		}
+	}
+
+	if len(srcNode["object"].([]interface{})) != 0 {
+		s := make([]string, 0, len(srcNode["object"].([]interface{})))
+		for _, val := range srcNode["object"].([]interface{}) {
+			if strVal, ok := val.(string); ok {
+				s = append(s, strVal)
+			}
+		}
+
+		// "" will be ignored, add it here if the input is [ "" ]
+		if len(s) == 0 {
```

Comment:
it seems that this condition always be false

Reply:
it will be effected when object = [""].